### PR TITLE
update

### PR DESCRIPTION
**Pull request by kemeng001**
update

### CODE REVIEW

File: src/src/main/java/StringManipulation.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import java.util.*;
+
+public class StringManipulation implements StringManipulationInterface {
+    private String s;
+
+    @Override
+    public void setString(String s) {
```
Comment (riteshsamal-ship) [inline src/src/main/java/StringManipulation.java:7]: The code checks for null or empty string in the count() and getSubStrings() methods, which is good. However, it doesn't handle null or empty string input in the setString() method. You might want to add a check for that.

File: src/src/main/java/StringManipulation.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    @Override
+    public String[] getSubStrings(int start, int end) {
+        if (s == null || s.length() == 0) {
+            return new String[0];
+        }
+        String[] words = s.split("\\s+");
+        if (start < 0 || end > words.length || start > end) {
+            return new String[0];
+        }
+        return Arrays.copyOfRange(words, start, end+1);
```
Comment (riteshsamal-ship) [inline src/src/main/java/StringManipulation.java:50]: The getSubStrings() method splits the string into words using \\s+ as the delimiter. It then returns a substring array based on the provided start and end indices. However, the end index is exclusive in Arrays.copyOfRange(), so you could use end instead of end+1 when copying the range

File: src/src/test/java/StringManipulationTest.java [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+    @BeforeEach
+    public void setUp() {
+        manipulatedstring = new StringManipulation();
+    }
+
+    @AfterEach
+    public void tearDown() {
+        manipulatedstring = null;
+    }
+
```
Comment (riteshsamal-ship) [inline src/src/test/java/StringManipulationTest.java:19]: You could add comments before each test case to explain what each one is verifying.

File: src/src/test/java/StringManipulationTest.java [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+        assertEquals(0, length);
+    }
+
+    @Test
+    public void testCount4() {
+        manipulatedstring.setString("SingleWord");
+        int length = manipulatedstring.count();
+        assertEquals(1, length);
+    }
+
```
Comment (riteshsamal-ship) [inline src/src/test/java/StringManipulationTest.java:47]: You could add some tests to handle if the string is empty or if after removing you end up with an empty string

File: src/src/test/java/StringManipulationTest.java [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+        manipulatedstring.setString("1");
+        assertEquals("1", manipulatedstring.removeNthCharacter(4, true));
+    }
+
+    @Test
+    public void testRemoveNthCharacter7() {
+        manipulatedstring.setString("123");
+        assertEquals("12", manipulatedstring.removeNthCharacter(3, false));
+    }
+
```
Comment (riteshsamal-ship) [inline src/src/test/java/StringManipulationTest.java:89]: You handle the testSubString cases really well but you may want to add exception handling for if the indices are out of bound.

**Review by riteshsamal-ship — APPROVED** [review]
